src/npc_attack.cpp

```diff
@@ -361,21 +434,34 @@ std::vector<npc_attack_rating> npc_attack_activate_item::all_evaluations( const
 
 void npc_attack_throw::use( npc &source, const tripoint &location ) const
 {
-    // WARNING! wielding the item invalidates the reference!
```

Comment:
this is still true, i dunno why you removed this comment

Reply:
Sorry, missed this comment. Now after trying to wield the weapon the reference isn't used anymore and the next action is recalculated next turn (likely attacking with that weapon if nothing has changed), creating a new attack instance with its own weapon reference.